### ISSUE TITLE
An error occurs when expanding abbreviations with the extension sass in VSCode 1.53 or higher.

### ISSUE THREAD

**Issue by mu60** [microsoft/vscode#117229]
![SnapCrab_● stylesass - test (ワークスペース) - Visual Studio Code_2021-2-22_20-57-13_No-00](https://user-images.githubusercontent.com/36182207/108706328-cd2cd400-7551-11eb-8834-cf91dbd4b8a4.jpg)

- VS Code Version: 1.53 or more
- OS Version: Windows10 Home

Steps to Reproduce:

1. Open the file with the extension sass.
2. Enter the abbreviation to perform the expansion.
3. The first deployment will be performed, but after the second deployment, the error "Failed to expand addreviation" will always occur and the deployment cannot be executed.
4. Regardless of the content of the abbreviation, an error will always occur in the second and subsequent expansions.
5. As long as I'm running VS Code, I keep getting errors.

Does this issue occur when all extensions are disabled?: No

If you do not enable the extension "Sass", you cannot use sass, so you cannot disable it.

**Comment by rzhao271** [comment]
Hi, does this issue still occur for you using the Insiders build?
https://code.visualstudio.com/insiders/

**Comment by mu60** [comment]
![SnapCrab_● stylesass - test (ワークスペース) - Visual Studio Code - Insiders_2021-2-23_9-48-52_No-00](https://user-images.githubusercontent.com/36182207/108788915-60025880-75bc-11eb-9778-6a4c4f3896cc.jpg)

I downloaded version 1.54.0 of insider.
However, the emmet error did not go away.

**Comment by mu60** [comment]
![SnapCrab_● stylesass - test - Visual Studio Code - Insiders_2021-2-23_9-54-23_No-00](https://user-images.githubusercontent.com/36182207/108789319-59281580-75bd-11eb-9711-0a25a8f37e45.jpg)

If I installed only the extension "sass" and all the key bindings were initialized, the emmet error did not occur.

**Comment by mu60** [comment]
![SnapCrab_● stylesass - test - Visual Studio Code_2021-2-23_10-19-24_No-00](https://user-images.githubusercontent.com/36182207/108790662-a063d580-75c0-11eb-85e0-e6d141d5825d.jpg)

In the stable version 1.53.2, if everything except the extension was initialized, the emmet error did not occur.
I changed the keybindings and no error occurred.
Something in your user settings may be affecting emmet.

**Comment by mu60** [comment]
![SnapCrab_settingsjson - test - Visual Studio Code - Insiders_2021-2-23_10-31-27_No-00](https://user-images.githubusercontent.com/36182207/108791400-51b73b00-75c2-11eb-945a-b51219fe37ee.jpg)

`{
    "workbench.colorTheme": "One Dark Pro",
    "editor.fontFamily": "Source Han Code JP",
    "editor.minimap.enabled": false,
    "emmet.extensionsPath": "D:/GoogleDrive/Documents/emmet",
    "emmet.triggerExpansionOnTab": true,
    "php.validate.executablePath": "C:/php-7.4.12/php.exe",
    "editor.parameterHints.enabled": false,
    "editor.acceptSuggestionOnEnter": "off",
    "explorer.confirmDelete": false,
    "editor.acceptSuggestionOnCommitCharacter": false,
    "emmet.showExpandedAbbreviation": "never",
    "editor.insertSpaces": false,
    "editor.quickSuggestions": false,
    "files.trimTrailingWhitespace": true,
    "files.trimFinalNewlines": true,
    "editor.autoClosingQuotes": "never",
    "editor.autoClosingBrackets": "never",
    "editor.autoSurround": "never",
    "files.insertFinalNewline": true,
	"editor.detectIndentation": false,
	"[html]": {
		"editor.defaultFormatter": "HookyQR.beautify"
	},
	"editor.occurrencesHighlight": false,
	"editor.codeLens": false,
	"editor.hover.enabled": false,
	"window.zoomLevel": 0,
	"html.autoClosingTags": false,
	"update.mode": "manual"
}
`

This is setting.json.
I feel that one of these is affecting Emmet.

**Comment by mu60** [comment]
The cause of the error has been determined.

`
// setting.json
{
    "emmet.extensionsPath": "D:/GoogleDrive/Documents/emmet"
}
`

First, I entered emmet.extensionsPath in setting.json.

`
// D:/GoogleDrive/Documents/emmet/snippet.json
{
	"variables": {
		"lang": "ja",
		"locale": "ja-JP"
	},
	"css": {
		"snippets": {
			"tal": "text-align: left",
			"bbn": "border-bottom: none",
			"brn": "border-right: none",
			"bln": "border-left: none",
			"btn": "border-top: none",
			"fww": "flex-wrap: wrap",
			"f": "flex: |",
			"mla": "margin-left: auto",
			"mra": "margin-right: auto",
			"mta": "margin-top: auto",
			"mba": "margin-bottom: auto",
			"ha": "height: auto",
			"wa": "width: auto",
			"fdc": "flex-direction: column",
			"lstn": "list-style-type: none",
			"bdn": "border: none",
			"bgl": "background: linear-gradient(|)"
		}
	}
}
`

And in snippet.json, the input of "|" was the cause of the error.
After removing the lines containing the "|", the Emmet error no longer occurs.
Confirmed with stable 1.53.2 and insider 1.54.0.

**Comment by rzhao271** [comment]
Interesting case. We did switch to a newer Emmet during 1.53, so the format of the snippet file changed a bit. For example, see https://github.com/emmetio/emmet/blob/master/snippets/css.json

**Comment by mu60** [comment]
Thank you.

`
"bgl": "background: linear-gradient(${1})"
`
By replacing "|" with "$ {1}", the error disappeared.

**Comment by rzhao271** [comment]
Actually, I just realized that this might be due to the parseSnippets call in vscode-emmet-helper. Maybe if we take that call off, your older format snippets will work again. More experimentation is needed.

**Comment by sergeche** [comment]
Yes, `|` is no longer supported. I’m trying to reduce the number of ways to do the same thing to simplify code and improve maintainability. Basically, it can be fixed by replacing `|` with `${1}`
